Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -2557,77 +2578,134 @@ private void HandleTickSize(object sender, IB.TickSizeEventArgs e)
         {
             Symbol symbol;
 
-            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol)) return;
+            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol))
+            {
+                return;
+            }
 
-            var tick = new Tick();
-            // in the event of a symbol change this will break since we'll be assigning the
-            // new symbol to the permtick which won't be known by the algorithm
-            tick.Symbol = symbol;
             var securityType = symbol.ID.SecurityType;
-            tick.Quantity = AdjustQuantity(securityType, e.Size);
-            tick.Time = GetRealTimeTickTime(symbol);
-
-            if (tick.Quantity == 0) return;
+            var quantity = AdjustQuantity(securityType, e.Size);
```

Comment:
I think `quantity` could also be reported as `-1` in some cases, should we add a check for it? `"...When IBApi::EWrapper::tickPrice and IBApi::EWrapper::tickSize are reported as -1, this indicates that there is no data currently available...."`

Reply:
Do you have a link for that quote?

Looking at the latest IB API docs for `tickPrice` here:
https://interactivebrokers.github.io/tws-api/interfaceIBApi_1_1EWrapper.html#ae851ec3a1e0fa2d0964c7779b0c89718
```
Market data tick price callback. Handles all price related ticks. 
Every tickPrice callback is followed by a tickSize. 
A tickPrice value of -1 or 0 followed by a tickSize of 0 indicates there is no data for this field currently available, whereas a tickPrice with a positive tickSize indicates an active quote of 0 (typically for a combo contract).
```
This was also confirmed by my logs during testing, where I have never seen a negative `tickSize` returned.

Reply:
Right, link @ https://interactivebrokers.github.io/tws-api/md_receive.html#market_data_return 3rd paragraph, I think its the case where there is no data available at all vs the field, I guess adding the check for -1 won't cause any harm?

Reply:
Ok, I'll add the normalization for negative quantity as well.
The Trade tick with non-positive quantity will be discarded anyway by `Tick.IsValid()`.